aten/src/ATen/PythonTorchFunctionTLS.cpp

```diff
@@ -43,7 +43,8 @@ const PythonTorchFunctionTLS& PythonTorchFunctionTLS::get_state() {
 }
 
 bool torch_function_mode_enabled() {
-  return PythonTorchFunctionTLS::stack_len() > 0;
+  return PythonTorchFunctionTLS::get_disabled_state() != TorchFunctionDisabledState::ALL_DISABLED &&
```

Comment:
it seems better to factor this into a helper function that tests if modes are enabled... what if you add another entry to the enum?